### PR TITLE
Add test cases for velocity chart task BE changes.

### PR DESCRIPTION
**Pull request by Alvilb**
# Issue:
task for velocity chart: https://app.asana.com/0/1202107027507930/1202165044911754/f
previous pr has incomplete unit tests https://github.com/framgia/sph-ut-uni-charts/pull/22

# Definition of Done
all test cases should pass
all BE changes related to velocity chart should be tested

### Note:
- new package is installed in `/backlog`

# Commands to run
- `docker-compose exec backlog yarn install`
- `docker-compose exec bff yarn install`
- `docker-compose restart backlog bff`
- `docker-compose exec bff yarn run test`
- `docker-compose exec backlog yarn run test`

# Route
n/a

# Attachments
bff test results
image

backlog test results
image

### CODE REVIEW

**Review by ziontavera — COMMENTED** [review]
Data has been deleted in Prisma DB. Please make it so that it only mocks the function and not delete actual data in database :bow:
Screenshot from 2022-05-12 12-09-18
Screenshot from 2022-05-12 12-09-26

**Comment by NielmarLaurente** [comment]
## Comment : 
- Test failed in `bff` 

image

**Review by ziontavera — APPROVED** [review]
LGTM :+1: 

Backlog:
Screenshot from 2022-05-12 16-03-52

Bff:
Screenshot from 2022-05-12 16-06-52

File: bff/src/test/controllers/IssueController.test.ts [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+const projectsResponse = {
+  id: 1,
+  name: 'name',
+  key: 'key',
+  project_id: 1,
+  provider_id: 1,
+  created_at: '2022-05-11T09:39:48.439Z',
+  updated_at: '2022-05-11T09:39:29.786Z'
+}
```
Comment (escbooster12-dev) [inline bff/src/test/controllers/IssueController.test.ts:67]: - please transfer this one to test directory. `tests/constants/issue.json`
- do the same in file **bff/src/utils/constants.ts**

**Review by escbooster12-dev — COMMENTED** [review]
- please update the description of each tests `describe`. make it more verbose.
- follow this blog https://markus.oberlehner.net/blog/naming-your-unit-tests-it-should-vs-given-when-then/

File: backlog/test/controllers/ProviderController.test.ts [hunk 2]
```diff
@@ -1,155 +1,212 @@
...
-    expect(res.statusCode).toEqual(200)
-    expect(res._getData()).toMatchObject({
-      user_id: 1,
-      name: 'backlog',
-      space_key: 'UNI-CHART',
-      api_key: 'apikey1234567890'
+  describe('if ID is not a number', () => {
+    it('should return a status of 400', () => {
+      req.params = { id: 'as' }
+      Controller.getProviderById(req, res)
```
Comment (escbooster12-dev) [inline backlog/test/controllers/ProviderController.test.ts:137]: - this one is repeating in **line:143** please move this `beforeEach` function
- there are also repeating in file **backlog/test/routes/ProviderRoute.test.ts**. do the same (move it to `beforeEach`) function.

**Review by escbooster12-dev — APPROVED** [review]
- tests are all working in both **bff** and **backlog** service
image

**Pull request merged**